crypto/cmp/cmp_msg.c

```diff
@@ -353,13 +354,20 @@ OSSL_CMP_MSG *ossl_cmp_certReq_new(OSSL_CMP_CTX *ctx, int type, int err_code)
             CMPerr(0, CMP_R_MISSING_PRIVATE_KEY);
             goto err;
         }
-        if ((crm = crm_new(ctx, type, OSSL_CMP_CERTREQID)) == NULL
-            || !OSSL_CRMF_MSG_create_popo(crm, privkey, ctx->digest,
-                                          ctx->popoMethod)
-            /* value.ir is same for cr and kur */
-            || !sk_OSSL_CRMF_MSG_push(msg->body->value.ir, crm))
+        if (crm == NULL) {
+            if ((local_crm = crm_new(ctx, type, OSSL_CMP_CERTREQID)) == NULL
```

Comment:
@DDvO can this function:crm_new export to user to creat a OSSL_CRMF_MSG?

Reply:
Yes, will do that...

Reply:
After having a closer look at `crm_new()` I don't see much point in exporting it 
because it is closely tied to the way CRMF structures are filled in by the vanilla CMP client implementation.
If you want to follow this you can use `crm_new()` indirectly via `OSSL_CMP_exec_certreq(ctx, req_type, NULL)`,
and if not, you have various exported `OSSL_CRMF_MSG_...` and `OSSL_CRMF_CERTTEMPLATE_...` functions to construct a custom `OSSL_CRMF_MSG`.
Or what specifically are you missing, @stardrift1?

Reply:
I just want to follow this ,but i just want a OSSL_CRMF_MSG created by the crm_new() function and then compute the popo of it locally and then tranfer it to a proxy use OSSL_CMP_try_certreq() do the cmp req, so I just want to use it directly but not use OSSL_CRMF_MSG_... and OSSL_CRMF_CERTTEMPLATE_... functions to construct a custom OSSL_CRMF_MSG.And in the  crm_new() fuction there are many action operate the CTX structure directly such as crit = ctx->setSubjectAltNameCritical ,when i creat the crmf myself,i can't operate the CTX structure directly right?

Reply:
@DDvO Hope you get what i mean

Reply:
All right, I've exported `crm_new()` under the name `OSSL_CMP_CTX_setup_CRM()`
(with one of its parameters streamlined).

Reply:
It's nice